### PR TITLE
Add threshold support for classifier models

### PR DESCRIPTION
**Pull request by mknitt**
This PR adds configurable decision threshold support to Flowcean's classifier models, enabling users to optimize the precision-recall tradeoff without retraining models.

Classifiers now support:
- Custom decision thresholds, which can be set at training time or adjusted after training is done
- Probability predictions using predict_proba() method for all classifiers
- Automatic threshold tuning to find optimal threshold for any metric

Usage Examples
```
# Train with custom threshold
learner = XGBoostClassifierLearner(threshold=0.3)  # Favor recall
model = learn_offline(data, learner, inputs, outputs)

# Change threshold after training
model.threshold = 0.7  # Favor precision

# Get probability predictions
probabilities = model.predict_proba(test_data)

# Automatically find optimal threshold
from flowcean.sklearn.metrics.classification import FBetaScore

best_threshold, results = tune_threshold(
    model, 
    validation_data, 
    inputs, 
    outputs, 
    metric=FBetaScore(beta=1.0)
)
model.threshold = best_threshold
```

### CODE REVIEW

File: src/flowcean/core/strategies/offline.py [hunk 1]
```diff
@@ -104,3 +104,151 @@ def evaluate_offline(
...
+    finally:
+        # Restore original threshold
+        model.threshold = original_threshold
+
+    logger.info(
+        "Best threshold: %.3f with %s = %.4f",
+        best_threshold,
+        metric.name,
+        best_score,
+    )
```
[the remark this answers is not in the document]
Comment (mknitt) [reply src/flowcean/core/strategies/offline.py]: removed

File: src/flowcean/core/strategies/offline.py [hunk 2]
```diff
@@ -104,3 +104,151 @@ def evaluate_offline(
             },
         )
     return Report(entries)
+
+
+def tune_threshold(
+    model: Model,
```
[the remark this answers is not in the document]
Comment (mknitt) [reply src/flowcean/core/strategies/offline.py]: done. `tune_threshold` now takes `ClassifierModel` instead of `Model`.

File: src/flowcean/core/model.py [hunk 3]
```diff
@@ -17,10 +17,22 @@ class Model(Named, Protocol):
     """Base class for models.
 
     A model is used to predict outputs for given inputs.
+
+    Optional Protocol Extensions:
+        For classification models that support probability predictions and
+        custom thresholds:
+        - Implement `_predict_proba` method to return class probabilities
+        - Implement `predict_proba` method (with preprocessing)
+        - Set `threshold` attribute to customize the decision boundary
```
[the remark this answers is not in the document]
Comment (mknitt) [reply src/flowcean/core/model.py]: done. `threshold` and the "Optional Protocol Extensions" comment block were removed from `Model`. A new `ClassifierModel(Model, Protocol)` was added at the bottom of the file

File: src/flowcean/sklearn/model.py [hunk 4]
```diff
@@ -30,20 +41,72 @@ def __init__(
...
         self._name = name
         self.estimator = estimator
         self.output_names = list(output_names)
+        self.threshold = threshold
+
+    def _predict_proba(
+        self,
+        input_features: pl.DataFrame | pl.LazyFrame,
+    ) -> pl.LazyFrame:
+        """Predict class probabilities (only for classifiers)."""
```
[the remark this answers is not in the document]
Comment (mknitt) [reply src/flowcean/sklearn/model.py]: done. 

- `SciKitModel` wraps any sklearn estimator, uses only `estimator.predict`.
- `SciKitClassifierModel(SciKitModel)` adds `threshold`, `predict_proba()`, and a `_predict()` that always applies the threshold.

**Review by ChristianWieck — APPROVED** [review]
